src/services/rgbpp.ts

```diff
@@ -298,9 +306,86 @@ export default class RgbppCollector extends BaseQueueWorker<IRgbppCollectRequest
     return null;
   }
 
+  async isIsomorphicTx(
+    btcTx: Transaction,
+    ckbTx: CKBComponents.RawTransaction,
+    validateCommitment?: boolean,
+  ): Promise<boolean> {
+    // Find the commitment from the btc_tx
+    const btcTxCommitment = tryGetCommitmentFromBtcTx(btcTx);
+    if (!btcTxCommitment) {
+      return false;
+    }
+
+    // Check inputs:
+    // 1. Find the last index of the type inputs
+    // 2. Check if all rgbpp_lock inputs can be found in the btc_tx.vin (regardless the position)
+    // 3. Check if the inputs contain at least one rgbpp_lock cell (as L1-L1 and L1-L2 transactions should have)
+    const inputs = await this.cradle.ckb.getInputCellsByOutPoint(ckbTx.inputs.map((input) => input.previousOutput!));
+    const lastTypeInputIndex = findLastIndex(inputs, (input) => !!input.cellOutput.type);
+    const anyRgbppLockInput = inputs.some((input) => isRgbppLock(input.cellOutput.lock));
+    if (!anyRgbppLockInput) {
+      return false;
+    }
+    const allInputsValid = inputs.every((input) => {
+      if (!input.cellOutput.type) {
+        return true;
+      }
+      if (!isRgbppLock(input.cellOutput.lock)) {
+        return true;
+      }
+      const rgbppLockArgs = unpackRgbppLockArgs(input.cellOutput.lock.args);
+      const matchingBtcInput = btcTx.vin.find(
+        (btcInput) => btcInput.txid === remove0x(rgbppLockArgs.btcTxid) && btcInput.vout === rgbppLockArgs.outIndex,
+      );
+      return !!matchingBtcInput;
+    });
+    if (!allInputsValid) {
+      return false;
+    }
+
+    // Check outputs:
+    // 1. Find the last index of the type outputs
+    // 2. Check if all type outputs are rgbpp_lock or btc_time_lock cells
+    // 4. Check if each rgbpp_lock cell has an isomorphic UTXO in the btc_tx.vout
+    // 5. Check if each btc_time_lock cell contains the corresponding btc_txid in the lock args
+    const lastTypeOutputIndex = findLastIndex(ckbTx.outputs, (output) => !!output.type);
+    const allOutputsValid = ckbTx.outputs.every((output) => {
+      if (!output.type) {
+        return true;
+      }
+      if (isRgbppLock(output.lock)) {
+        const rgbppLockArgs = unpackRgbppLockArgs(output.lock.args);
+        const btcTxId = remove0x(rgbppLockArgs.btcTxid);
+        if (btcTxId === RGBPP_TX_ID_PLACEHOLDER) {
+          return true;
+        }
```

Comment:
```suggestion
    const lastTypeOutputIndex = findLastIndex(ckbTx.outputs, (output) => !!output.type);
```
### Suggestion
Delete the above check, to allow for the case where there are `no outputs with type_script` in the transaction.

scenario: No type_script in the ckb_tx_outputs means burning the rgbpp-asset or recollect the capacity in the type_scripts_input_cells.